test/framework/src/main/java/org/elasticsearch/test/ClusterServiceUtils.java

```diff
@@ -122,11 +122,20 @@ public static ClusterService createClusterService(ThreadPool threadPool, Cluster
 
     public static ClusterService createClusterService(ThreadPool threadPool, DiscoveryNode localNode, ClusterSettings clusterSettings) {
         Settings settings = Settings.builder().put("node.name", "test").put("cluster.name", "ClusterServiceTests").build();
+        return createClusterService(threadPool, localNode, settings, clusterSettings);
+    }
+
+    public static ClusterService createClusterService(
+        ThreadPool threadPool,
+        DiscoveryNode localNode,
+        Settings nodeSettings,
+        ClusterSettings clusterSettings
+    ) {
         ClusterService clusterService = new ClusterService(
-            settings,
+            nodeSettings,
             clusterSettings,
             threadPool,
-            new TaskManager(settings, threadPool, Collections.emptySet(), Tracer.NOOP)
+            new TaskManager(nodeSettings, threadPool, Collections.emptySet(), Tracer.NOOP)
```

Comment:
I would suggest the following refactoring here:
```
public static ClusterService createClusterService(ThreadPool threadPool, DiscoveryNode localNode, ClusterSettings clusterSettings) {
        return createClusterService(threadPool, localNode, Settings.EMPTY, clusterSettings);
}

public static ClusterService createClusterService(
        ThreadPool threadPool,
        DiscoveryNode localNode,
        Settings providedSettings,
        ClusterSettings clusterSettings
    ) {
        Settings settings = Settings.builder()
            .put("node.name", "test")
            .put("cluster.name", "ClusterServiceTests")
            .put(providedSettings)
            .build();
        ClusterService clusterService = new ClusterService(
            settings,
            clusterSettings,
            threadPool,
            new TaskManager(settings, threadPool, Collections.emptySet(), Tracer.NOOP)
        );
        ....
```
This way you do not need to leak into your test the `node.name` and `cluster.name` but you can still provide the additional settings you need. 

As you seeI avoided calling it `nodeSettings` because in the `APMIndexTemplateRegistry` this name represents something else and that confused me in the beginning.

Reply:
Thanks, I didn't know could do that with the settings. Updated!